scalyr_agent/monitor_utils/k8s.py

```diff
@@ -210,6 +209,156 @@ class KubeletApiException(Exception):
+        :rtype: K8sNamespaceFilter
+        """
+        return K8sNamespaceFilter(global_include=["*"], global_ignore=[])
+
+    @staticmethod
+    def default_value():
+        """Returns the filter created by the configuration when all the defaults are used.
+        :return: The default filter.
+        :rtype: K8sNamespaceFilter
+        """
+        return K8sNamespaceFilter(
+            global_include=["*"],
+            global_ignore=Configuration.DEFAULT_K8S_IGNORE_NAMESPACES,
+        )
+
+    @staticmethod
+    def from_config(global_config=None, local_config=None):
+        """Convenience method for created a filter object based on the global configuration file and
+        a monitor's configuration.
+
+        :param global_config: The global configuraiton object.  This is used to retrieve the global include and ignore
+            list.
+        :param local_config: The local configuration object, if any.  If not None, this is used to retrieve the
+            local ignore list.
+        :type global_config: Configuration
+        :type local_config: MonitorConfiguration
+        :return: The filter.
+        :rtype: K8sNamespaceFilter
+        """
+        global_ignore = global_config.k8s_ignore_namespaces
+        global_include = global_config.k8s_include_namespaces
+
+        if local_config is not None:
+            local_ignore = local_config.get("k8s_ignore_namespaces")
+        else:
+            local_ignore = None
+        return K8sNamespaceFilter(
+            global_include=global_include,
+            global_ignore=global_ignore,
+            local_ignore=local_ignore,
+        )
+
+    @staticmethod
+    def __namespace_set(namespace_list):
+        """Creates a set containing the names in the specified list.  This also (for convenience) strips out
+        leading and trailing whitespace.
+
+        :param namespace_list: The list of namespaces
+        :type namespace_list: [six.text_type]
+        :return: The set
+        :rtype: set
+        """
+        result = set()
+        if namespace_list is None:
+            return result
+
+        for x in namespace_list:
+            result.add(x.strip())
+
+        return result
+
+    def __contains__(self, key):
+        """Returns true if the given namespace passes the filter.  This means the namespace is in the whitelist
+        (or the whitelist is *) and does not appear in the blacklist.
+
+        :param key: The namespace to test
+        :type key: six.text_type
+        :return: True if the namespace passes the filter.
+        :rtype: bool
+        """
+        if self.__whitelist is not None:
+            return key in self.__whitelist
+        else:
+            return key not in self.__blacklist
+
+    def __str__(self):
+        if self.__whitelist is not None:
+            return "include_only=%s" % ",".join(self.__whitelist)
+        else:
+            return "exclude=%s" % ",".join(self.__blacklist)
+
+    def __eq__(self, other):
```

Comment:
In python 2:

> The truth of x==y does not imply that x!=y is false. Accordingly, when defining __eq__(), one should also define __ne__() so that the operators will behave as expected.

In python 3:

> By default, __ne__() delegates to __eq__() and inverts the result unless it is NotImplemented. 

To ensure consistency between 2 and 3, we should define `__ne__`.  [This SO post](https://stackoverflow.com/a/30676267) suggests it's better in python2 to define `__ne__` as `not ==` e.g.

```
def __ne__(self, other):
        return not self == other
```

Reply:
Thanks, I didn't know about that nuance.  Implemented as you suggested.

Reply:
I had to look it up myself, but I had a vague memory that python2 had different behavior.